xds/src/main/java/io/grpc/xds/client/ControlPlaneClient.java

```diff
@@ -391,26 +445,43 @@ private void handleRpcStreamClosed(Status error) {
         // has never been initialized.
         retryBackoffPolicy = backoffPolicyProvider.get();
       }
-      // FakeClock in tests isn't thread-safe. Schedule the retry timer before notifying callbacks
-      // to avoid TSAN races, since tests may wait until callbacks are called but then would run
-      // concurrently with the stopwatch and schedule.
-      long elapsed = stopwatch.elapsed(TimeUnit.NANOSECONDS);
-      long delayNanos = Math.max(0, retryBackoffPolicy.nextBackoffNanos() - elapsed);
-      rpcRetryTimer = syncContext.schedule(
-          new RpcRetryTask(), delayNanos, TimeUnit.NANOSECONDS, timeService);
-
-      checkArgument(!error.isOk(), "unexpected OK status");
-      String errorMsg = error.getDescription() != null
-          && error.getDescription().equals(CLOSED_BY_SERVER)
-              ? "ADS stream closed with status {0}: {1}. Cause: {2}"
-              : "ADS stream failed with status {0}: {1}. Cause: {2}";
-      logger.log(
-          XdsLogLevel.ERROR, errorMsg, error.getCode(), error.getDescription(), error.getCause());
+
+      boolean shouldTryFallback = !responseReceived && !shutdown;
+
+      if (!shutdown) {
```

Comment:
`shutdown == true` implies `closed == true`, so how do we get here with a value of anything but `shutdown == false`?

Reply:
You're right.  Also figured out that in the case of shutting down where it was skipping the method it didn't need to do anything anyway.